include/votca/tools/calculator.h

```diff
@@ -77,20 +77,40 @@ class Calculator {
    * @param output stream
    */
   void DisplayOptions(std::ostream &out);
+
+  /**
+   * \brief Loads default options stored in VOTCASHARE
+   */
+  void LoadDefaults(std::string package = "tools");
+
   /**
    * \brief Updates options with default options stored in VOTCASHARE
    *
    * If a value is not given or tag is not present and at the same time
    * a default value exists in the corresponding XML file in VOTCASHARE
    * a tag is created and/or a default value is assigned to it
    */
-  void UpdateWithDefaults(Property &options, std::string package = "tools");
+  void UpdateWithUserOptions(const Property &options);
+
+  /**
+   * \brief Load the default options and merge them with the user input
+   *
+   * Defaults are overwritten with user input
+   */
+  void LoadDefaultsAndUpdateWithUserOptions(std::string package,
+                                            const Property &options) {
+    LoadDefaults(package);
+    UpdateWithUserOptions(options);
+  }
 
  protected:
   Index _nThreads;
   bool _maverick;
 
-  void AddDefaults(Property &p, const Property &defaults);
+  void AddDefaults(const Property &p, Property &defaults);
```

Comment:
why is defaults non const, the name suggest that p should be modified.

Reply:
Actually, it is the other way around we are adding the properties `p` to the defaults. The name is misleading and I am going to change it

Reply:
I have renamed `AddDefaults` to `OverwriteDefaultsWithUserInput`